src/xenia/kernel/XLiveAPI.cpp

```diff
@@ -473,6 +473,7 @@ std::unique_ptr<HTTPResponseObjectJSON> XLiveAPI::RegisterPlayer() {
 
   // User index hard-coded
   player.XUID(kernel_state()->user_profile((uint32_t)0)->xuid());
+  player.Gamertag(kernel_state()->user_profile((uint32_t)0)->name());
```

Comment:
Truncate the gamertag to 15 characters (max len) before sending it to server for storing.